src/error.rs

```diff
@@ -58,6 +58,7 @@ impl Error {
         match self.err.code {
             ErrorCode::Message(_) => Category::Data,
             ErrorCode::Io(_) => Category::Io,
+            ErrorCode::ControlCharacterWhileParsingString |
```

Comment:
This error is a syntax error so it should be categorized as `Category::Syntax` not `Category::Eof`.

Reply:
Oops, thought I had that in the syntax category.  Done.